lib/ecl/ecl_grid.cpp

```diff
@@ -7111,4 +7112,53 @@ void ecl_grid_export_index(const ecl_grid_type * grid, int * global_index, int *
       }
 }
 
+//Note: index_size must equal allocated size of output
+void ecl_grid_export_data_as_int( ecl_grid_type * grid, int index_size, int * global_index, ecl_kw_type * kw, int * output) {
+  int * input = ecl_kw_get_int_ptr(kw);
```

Comment:
The size of the index should not be relevant, the important question to if on is the size of keyword. Before calling this function we must check the size of the keyword, and then send either active_index of global_index; something like:
```python
if len(kw) == grid.get_global_size():
   index = index_frame.get_global
elif len(kw) == grid.get_num_actiev():
   index = index_frame.get_active()
else:
   raise ValueError("The keyword must have a 3D compatible length")
```
and then the C++ code could look like:
```C++
ecl_grid_export_data_as_xxx(, ... int index_size, const int * data_index, .... )
...
for (int i=0; i < index_size; i++) {
     int di = data_index[i];
     if (di >= 0)
         output[i] = input[di]
}
```